### PR TITLE
OpenShift refactor

### PR DESCRIPTION
**Pull request by natlibfi-jonollil**
* Updated logic
* Updated tests

### CODE REVIEW

File: .github/workflows/melinda-node-tests.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    steps:
+    - name: Checkout the code
+      uses: actions/checkout@v2
+    - name: nodejsscan scan
+      id: njsscan
+      uses: ajinabraham/njsscan-action@master
+      with:
+        args: '.'
+
```
Comment (ammsalme) [inline .github/workflows/melinda-node-tests.yml:45]: Should we have license-scan here also?

File: src/transform/convert/generate0XXFields.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          });
+        }
+      }
+    });
+}
+
+export function handle037(marcRecord) {
+  marcRecord.get(/^037$/u).forEach(field => {
+    field.subfields.push({ // eslint-disable-line functional/immutable-data
+      code: '5', value: 'HELME<KEEP>'
```
Comment (ammsalme) [inline src/transform/convert/generate0XXFields.js:23]: This is wrong: it should be:
code: '5', value: 'HELME'

File: src/transform/convert/generate6XXFields.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export function handleTerms(marcRecord) {
```
Comment (ammsalme) [inline src/transform/convert/generate6XXFields.js:1]: Why are we transferring subject headings from 'kaunokki' to 'ysa'? Why are we transferring genre/form headings from 'kaunokki' and 'bella' to 'slm/fin' and 'slm/fin'?
- YSA is not in use anymore
- kaunokki/bella and ysa or slm probably are NOT one to one matches?

File: src/validate/index.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+export default async () => {
+  const validate = validateFactory([
+    await FieldsPresent([/^(020|022|024)$/u]), // eslint-disable-line
+    await FieldsPresent([/^336$/u, /^337$/u, /^338$/u]),
+    await FieldExclusion([
+      /^(001|091|092|093|094|095|256|533|574|575|576|577|578|599)$/u, // eslint-disable-line prefer-named-capture-group
+      {tag: /^264$/u, subfields: [{code: /^a$/u, value: /^\[.*\]$/u}]},
+      {tag: /^650$/u, subfields: [{code: /^a$/u, value: /^overdrive$/ui}]},
+      {tag: /^(648|650|651|655)$/u, subfields: [{code: /^2$/u, value: /^(ysa|musa|allars|cilla)$/u}]}, // eslint-disable-line prefer-named-capture-group
```
Comment (ammsalme) [inline src/validate/index.js:49]: Here we exclude subject headings from ysa, including those created earlier from kaunokki subject headeings.

File: test-fixtures/transform/02/outputSucces.json [hunk 5]
```diff
@@ -0,0 +1,678 @@
...
+          "ind2": " ",
+          "subfields": [
+            {
+              "code": "a",
+              "value": "Internet-yhteys, www-selain ; lukuohjelma."
+            }
+          ]
+        },
+        {
+          "tag": "540",
```
Comment (ammsalme) [inline test-fixtures/transform/02/outputSucces.json:232]: 540 with "Käyttöoikeus Helmet-kirjastokortilla" should have subfield '$5 HELME' , or preferably be dropped.

File: test-fixtures/transform/02/outputSucces.json [hunk 6]
```diff
@@ -0,0 +1,678 @@
...
+              "value": "Painettu:"
+            },
+            {
+              "code": "z",
+              "value": "9789511321224"
+            }
+          ]
+        },
+        {
+          "tag": "856",
```
Comment (ammsalme) [inline test-fixtures/transform/02/outputSucces.json:325]: Ellibs-librayr 856 should have subfield `$5 HELME` or preferably be dropped.

**Review by ammsalme — CHANGES_REQUESTED** [review]
License-scan? 
Also some comments about weirdness in data conversion, but were broken already, I guess.

File: src/transform/convert/generate6XXFields.js (hunk 3 above)
Comment (natlibfi-jonollil) [reply src/transform/convert/generate6XXFields.js:1]: Kaunokki => sml/fin ok
Bella => sml/swe ok
Kaunokki => ysa => dropped ok

**Review by ammsalme — APPROVED** [review]

**Review by ammsalme — APPROVED** [review]

**Pull request merged**